namerd/iface/interpreter-thrift/src/test/scala/io/buoyant/namerd/iface/ObserverCacheTest.scala

```diff
@@ -0,0 +1,124 @@
+package io.buoyant.namerd.iface
+
+import com.twitter.io.Buf
+import com.twitter.util._
+import io.buoyant.namerd.iface.ThriftNamerInterface.{Stamp, Observer}
+import org.scalatest.FunSuite
+
+class ObserverCacheTest extends FunSuite {
+
+  case class TestObserver[T](obs: Observer[T], event: Event[Try[T]] with Witness[Try[T]], closed: Var[Boolean])
+
+  def testObserver[T](): TestObserver[T] = {
+    val event = Event[Try[T]]
+    val closed = Var(false)
+    val obs = new Observer[T] {
+      var i = 0
+
+      protected[this] def nextStamp(): Stamp = {
+        val stamp = Buf.Utf8(i.toString)
+        i += 1
+        stamp
+      }
+
+      protected[this] val updater: Closable = {
+        val eventClosable = event.respond(update)
+        Closable.all(
+          eventClosable,
+          Closable.make { _ => closed() = true; Future.Unit }
+        )
+      }
+    }
+    TestObserver(obs, event, closed)
+  }
+
+  test("observer cache caches") {
+
+    var count = 0
+
+    val cache = new ObserverCache[String, String](1, 0)({ _ =>
```

Comment:
Ah, nevermind, I now see the "observer cache reactivates observations" test below.